zipline/algorithm.py

```diff
@@ -490,101 +476,28 @@ def get_generator(self):
-        if isinstance(source, list):
-            if overwrite_sim_params:
-                warnings.warn("""List of sources passed, will not attempt to extract start and end
- dates. Make sure to set the correct fields in sim_params passed to
- __init__().""", UserWarning)
-                overwrite_sim_params = False
-        elif isinstance(source, pd.DataFrame):
-            # if DataFrame provided, map columns to sids and wrap
-            # in DataFrameSource
-            copy_frame = source.copy()
-            copy_frame.columns = self._write_and_map_id_index_to_sids(
-                source.columns, source.index[0],
-            )
-            source = DataFrameSource(copy_frame)
-
-        elif isinstance(source, pd.Panel):
-            # If Panel provided, map items to sids and wrap
-            # in DataPanelSource
-            copy_panel = source.copy()
-            copy_panel.items = self._write_and_map_id_index_to_sids(
-                source.items, source.major_axis[0],
-            )
-            source = DataPanelSource(copy_panel)
-
-        if isinstance(source, list):
-            self.set_sources(source)
-        else:
-            self.set_sources([source])
-
-        # Override sim_params if params are provided by the source.
-        if overwrite_sim_params:
-            if hasattr(source, 'start'):
-                self.sim_params.period_start = source.start
-            if hasattr(source, 'end'):
-                self.sim_params.period_end = source.end
-            # Changing period_start and period_close might require updating
-            # of first_open and last_close.
-            self.sim_params.update_internal_from_env(
-                env=self.trading_environment
-            )
-
-        # The sids field of the source is the reference for the universe at
-        # the start of the run
-        self._current_universe = set()
-        for source in self.sources:
-            for sid in source.sids:
-                self._current_universe.add(sid)
-        # Check that all sids from the source are accounted for in
-        # the AssetFinder. This retrieve call will raise an exception if the
-        # sid is not found.
-        for sid in self._current_universe:
-            self.asset_finder.retrieve_asset(sid)
+        self.ensure_clock()
 
         # force a reset of the performance tracker, in case
         # this is a repeat run of the algorithm.
         self.perf_tracker = None
 
         # create zipline
-        self.gen = self._create_generator(self.sim_params)
-
-        # Create history containers
-        if self.history_specs:
-            self.history_container = self.history_container_class(
-                self.history_specs,
-                self.current_universe(),
-                self.sim_params.first_open,
-                self.sim_params.data_frequency,
-                self.trading_environment,
-            )
+        self.gen = self.get_generator()
```

Comment:
If `self.gen` is only used in this method, can we make it a local instead of an instance attribute?